cgroups/fs/apply_raw.go

```diff
@@ -88,6 +91,93 @@ func Apply(c *cgroups.Cgroup, pid int) (map[string]string, error) {
 	return paths, nil
 }
 
+// Stop stops all processes in the current cgroup
+func Stop(c *cgroups.Cgroup) error {
+	d, err := getCgroupData(c, 0)
+	if err != nil {
+		return err
+	}
+
+	// first we send SIGTERM to all tasks inside the cgroup
+	if err := Kill(c, syscall.SIGTERM); err != nil {
+		return err
+	}
+
+	// if failed to stop with only SIGTERM, we send SIGKILL
+	if err := waitStop(d, 100*time.Millisecond, 5*time.Second); err != nil {
+		if err := Kill(c, syscall.SIGKILL); err != nil {
+			return err
+		}
+		return waitStop(d, 100*time.Millisecond, -1*time.Second)
```

Comment:
So if i understand it correctly, the new strategy is:
- send SIGTERM to all processes
- wait for a while ( here it checks every 100 ms for up to 5s ). Do you think 5s is too small / too large ? Should it be configurable so that the process/user can specify its own waiting time ?
- if the processes do not exit after the timeout, then we send SIGKILL, and then wait for a while
- if the processes still do not exit after the timeout here, bubble up errors to user

Also `waitStop` backoff is constant here, do we need it to be exponential like in cgroup removal ?